api/src/modules/permissions/services/permissions.service.ts

```diff
@@ -1,23 +1,50 @@
 import { Injectable } from '@nestjs/common';
-import { DataService } from '../../data/services/data.service';
+import { DataService } from '../../data';
 import {
-	GetPermissionsDocument,
-	GetPermissionsQuery,
-	GetPermissionsQueryVariables,
-} from '../../shared/generated/graphql-db-types-hetarchief';
+	PermissionQueryTypes,
+	PERMISSIONS_QUERIES,
+} from '../permissions.consts';
+import { PermissionInfo } from '../permissions.types';
 
 @Injectable()
 export class PermissionsService {
 	constructor(private dataService: DataService) {}
 
-	public async getPermissions(): Promise<
-		GetPermissionsQuery['users_permission']
-	> {
+	public async getPermissions(): Promise<PermissionInfo[]> {
 		const response = await this.dataService.execute<
-			GetPermissionsQuery,
-			GetPermissionsQueryVariables
-		>(GetPermissionsDocument);
+			PermissionQueryTypes['GetPermissionsQuery']
+		>(PERMISSIONS_QUERIES[process.env.DATABASE_APPLICATION_TYPE]);
 
-		return response.users_permission;
+		if (
+			(response as PermissionQueryTypes['GetPermissionsQueryAvo'])
+				?.users_permissions
+		) {
```

Comment:
Why type-check the output instead of use `DATABASE_APPLICATION_TYPE` in a switch-case with casts? Eventually the nomenclature will align and the output will remain ambiguous.

Reply:
i convert the types to a uniform format, so we don't need to deal with the avo vs hetarchief data structure on the client.
* users_permissions vs users_permission
* avo doesn't have a name property
* label is used for the name property in avo

I'm not sure well be able to output these in a generic way without casting them to a specific type.
The switch could be useful when we add more apps, i agree.